core/amber/src/main/scala/edu/uci/ics/texera/web/resource/dashboard/user/quota/UserQuotaResource.scala

```diff
@@ -49,6 +60,52 @@ object UserQuotaResource {
       eid: UInteger
   )
 
+  def getFolderSize(folder: String): Long = {
+    val path = Paths.get(folder)
+
+    val walk = Files.walk(path)
+    try {
+      walk
+        .filter(Files.isRegularFile(_))
+        .mapToLong(p => Files.readAttributes(p, classOf[BasicFileAttributes]).size())
+        .sum()
+    } finally {
+      walk.close()
+    }
+  }
+
+  def getUserDatasetSize(uid: UInteger): Long = {
+    val datasetFolder = "./user-resources/datasets"
```

Comment:
Avoid defining such string literal in the function. To know the Path of the dataset given a did, you can use PathUtils.getDatasetPath. See `core/amber/src/main/scala/edu/uci/ics/texera/web/resource/dashboard/user/dataset/utils/PathUtils.scala`